youtube_client_adapter.py

```diff
@@ -0,0 +1,81 @@
+from youtube_video_metadata import YoutubeVideoMetadata
+
+
+class YoutubeClientAdapter:
+    def __init__(self, client, api_key: str) -> None:
+        self.client = client
+        self.api_key = api_key
+
+    def get_channel_videos(self, channel_id: str) -> list[YoutubeVideoMetadata]:
```

Comment:
TODO: add tests

Reply:
Note: static typing with `list[T]` requires Python version 3.9+. This can by changed to use the `List` type from the `typing` module if needed